edgedb-protocol/src/descriptors.rs

```diff
@@ -620,20 +867,38 @@ impl Decode for NamedTupleTypeDescriptor {
         ensure!(buf.remaining() >= 19, errors::Underflow);
```

Comment:
These seem somewhat brittle as there's no easy way to validate this length is correct and it doesn't account for the minimum difference in length between proto versions. Would it be more robust to use `decode` for everything and remove these higher-level length checks?

Reply:
Agree, I'd leave this for a future cleanup though.